contracts/Comptroller/Diamond/facets/PolicyFacet.sol

```diff
@@ -359,6 +363,77 @@ contract PolicyFacet is IPolicyFacet, XVSRewardsHelper {
         }
     }
 
+    /**
+     * @notice Executes a flashLoan operation with the specified assets and amounts.
+     * @dev Transfer the specified assets to the receiver contract and ensures that the total repayment (amount + fee)
+     *      is returned by the receiver contract after the operation for each asset. The function performs checks to ensure the validity
+     *      of parameters, that flashLoans are enabled for the given assets, and that the total repayment is sufficient.
+     *      Reverts on invalid parameters, disabled flashLoans, or insufficient repayment.
+     * @param receiver The address of the contract that will receive the flashLoan and execute the operation.
+     * @param vTokens The addresses of the vToken assets to be loaned.
+     * @param underlyingAmounts The amounts of each underlying asset to be loaned.
+     * @param param The bytes passed in the executeOperation call.
+     * custom:requirements
+     *      - `vTokens.length` must be equal to `underlyingAmounts.length`.
+     *      - `vTokens.length` and `underlyingAmounts.length` must not be zero.
+     *      - The `receiver` address must not be the zero address.
+     *      - FlashLoans must be enabled for each asset.
+     *      - The `receiver` contract must repay the loan with the appropriate fee.
+     * custom:reverts
+     *      - Reverts with `Invalid flashLoan params` if parameter checks fail.
+     *      - Reverts with `FlashLoan not enabled` if flashLoans are disabled for any of the requested assets.
+     *      - Reverts with `Execute flashLoan failed` if the receiver contract fails to execute the operation.
+     *      - Reverts with `Insufficient reypayment balance` if the repayment (amount + fee) is insufficient after the operation.
+     */
+    function executeFlashLoan(
```

Comment:
I understand this is a combination of `flashLoanSimple` and `flashLoan`: allowing to borrow funds from different markets, but not allowing to keep an open position, for example.

I would add the following features/parameters to this function:
* `address onBehalfOf`: to allow the sender of the function to open the loan on behalf of someone else (who previously has to approve the sender as a valid delegate. Out of the scope)
* `uint256[] memory modes`: how each borrowed asset is handled if not repaid immediately
  * 0: No debt (classic flash loan). The amount must be repaid within the same transaction, or it reverts. What we have in the current implementation
  * 1: A debt position is opened

Reply:
https://github.com/VenusProtocol/venus-protocol/pull/545/commits/41390b3a31e870a50b9e76b6c804e2ff17c6d4a6